packages/db-objects-redis/src/lib/objects/objectsInRedisClient.ts

```diff
@@ -2437,6 +2460,80 @@ export class ObjectsInRedisClient {
         );
     }
 
+    private async _subscribeFile(id: string, pattern: string) {
+        if (!this.sub) {
+            throw new Error(ERRORS.ERROR_DB_CLOSED);
+        }
+        if (Array.isArray(pattern)) {
+            for (let p = 0; p < pattern.length; p++) {
+                const fileId = this.getFileId(id, pattern[p], false);
+                this.log.silly(`${this.namespace} redis psubscribe ${fileId}`);
+                if (this.sub) {
+                    await this.sub.psubscribe(fileId);
+                    this.userSubscriptions[fileId] = true;
+                }
+            }
+        } else {
+            const fileId = this.getFileId(id, pattern, false);
+            this.log.silly(`${this.namespace} redis psubscribe ${fileId}`);
+            await this.sub.psubscribe(fileId);
+            this.userSubscriptions[fileId] = true;
+        }
+    }
+
+    private async _unsubscribeFile(id: string, pattern: string) {
+        if (!this.sub) {
+            throw new Error(ERRORS.ERROR_DB_CLOSED);
+        }
+        if (Array.isArray(pattern)) {
+            for (let p = 0; p < pattern.length; p++) {
+                const fileId = this.getFileId(id, pattern[p], false);
+                this.log.silly(`${this.namespace} redis punsubscribe ${fileId}`);
+                if (this.sub) {
+                    await this.sub.punsubscribe(fileId);
+                    if (this.userSubscriptions[fileId] !== undefined) {
+                        delete this.userSubscriptions[fileId];
+                    }
+                }
+            }
+        } else {
+            this.log.silly(`${this.namespace} redis punsubscribe ${this.objNamespace}${pattern}`);
+            const fileId = this.getFileId(id, pattern, false);
+            await this.sub.punsubscribe(fileId);
+            if (this.userSubscriptions[fileId] !== undefined) {
+                delete this.userSubscriptions[fileId];
+            }
+        }
+    }
+
+    subscribeUserFile(id: string, pattern: string, options?: CallOptions): Promise<void> {
+        return new Promise((resolve, reject) => {
+            utils.checkObjectRights(this, null, null, options, 'list', err => {
+                if (err) {
+                    reject(err);
+                } else {
+                    return this._subscribeFile(id, pattern)
```

Comment:
Formally it is subscribing anything on the "system" sub client and NOT on the user subscriptions client!

So in fact the method tells "subscribe for user" but is really subscribing on system client ... Is this intended?

Reply:
Actually vice-versa. Only on user:
![image](https://user-images.githubusercontent.com/4582016/167372227-5b46e56a-5ef8-48a1-bb55-304562a7cdfb.png)

https://github.com/ioBroker/ioBroker.js-controller/blob/af0f5bd5bca69d177ba82325f93292643e955315/packages/db-objects-redis/src/lib/objects/objectsInRedisClient.ts#L2479

Reply:
Ahhh then I got the code wrong ... ok